aries_cloudagent/messaging/jsonld/create_verify_data.py

```diff
@@ -38,53 +42,66 @@ def _canonize_document(doc):
         )
 
-    if "created" not in signature_options:
-        signature_options["created"] = datetime.datetime.now(
-            datetime.timezone.utc
-        ).strftime("%Y-%m-%dT%H:%M:%SZ")
-
-    if (
-        "type" not in signature_options
-        or signature_options["type"] != "Ed25519Signature2018"
-    ):
-        signature_options["type"] = "Ed25519Signature2018"
-
+    signature_options["created"] = signature_options.get("created", _created_at())
     [expanded] = jsonld.expand(data)
     framed = jsonld.compact(
         expanded, "https://w3id.org/security/v2", {"skipExpansion": True}
     )
 
     # Detect any dropped attributes during the expand/contract step.
-    if len(data) != len(framed):
-        raise DroppedAttributeError("Extra Attribute Detected")
-    if (
-        "proof" in data
-        and "proof" in framed
-        and len(data["proof"]) != len(framed["proof"])
-    ):
-        raise DroppedAttributeError("Extra Attribute Detected")
-    if (
-        "credentialSubject" in data
-        and "https://www.w3.org/2018/credentials#credentialSubject" in framed
-        and len(data["credentialSubject"])
-        != len(framed["https://www.w3.org/2018/credentials#credentialSubject"])
-    ):
-        raise DroppedAttributeError("Extra Attribute Detected")
-
-    canonized_signature_options = _canonize_signature_options(signature_options)
-    hash_of_canonized_signature_options = _sha256(canonized_signature_options)
-    canonized_document = _canonize_document(framed)
-    hash_of_canonized_document = _sha256(canonized_document)
-
-    return (framed, hash_of_canonized_signature_options + hash_of_canonized_document)
+    if len(data) > len(framed):
+        # > check indicates dropped attrs < is a different error
+        # attempt to collect error report data
+        for_diff = jsonld.compact(expanded, data.get("@context"))
+        dropped = set(data.keys()) - set(for_diff.keys())
+        raise DroppedAttributeError(
+            f"{dropped} attributes dropped. "
+            " Provide definitions in context to correct."
+        )
+    # Check proof for dropped attributes
+    attr = [
+        ("proof", "proof"),
+        ("credentialSubject", "https://www.w3.org/2018/credentials#credentialSubject"),
+    ]
+    data_context = data.get("@context")
+    for mapping in attr:
+        data_attribute = data.get(mapping[0], {})
+        frame_attribute = framed.get(mapping[1], {})
+        if len(data_attribute) > len(frame_attribute):
+            for_diff = jsonld.compact(expanded, data_context)
+            for_diff_attribute = for_diff.get(mapping[1], {})
+            dropped = set(data_attribute.keys()) - set(for_diff_attribute.keys())
+            raise DroppedAttributeError(
+                f"in {mapping[0]}, {dropped} attributes dropped. "
+                "Provide definitions in context to correct."
+            )
+
+    cannonized_signature_options = _canonize_signature_options(signature_options)
```

Comment:
canon not cannon - it's ecclesiastical canon not martial cannon

Reply:
Ahh! make peace, not war! I will correct that.